components/eamxx/src/physics/mam/eamxx_mam_optics_process_interface.cpp

```diff
@@ -0,0 +1,107 @@
+#include <physics/mam/eamxx_mam_optics_process_interface.hpp>
+#include <share/property_checks/field_lower_bound_check.hpp>
+#include <share/property_checks/field_within_interval_check.hpp>
+
+#include "scream_config.h" // for SCREAM_CIME_BUILD
+
+#include <ekat/ekat_assert.hpp>
+
+namespace scream
+{
+
+MAMOptics::MAMOptics(
+    const ekat::Comm& comm,
+    const ekat::ParameterList& params)
+  : AtmosphereProcess(comm, params),
+    aero_config_() {
+}
+
+AtmosphereProcessType MAMOptics::type() const {
+  return AtmosphereProcessType::Physics;
+}
+
+std::string MAMOptics::name() const {
+  return "mam4_optics";
+}
+
+void MAMOptics::set_grids(const std::shared_ptr<const GridsManager> grids_manager) {
+  using namespace ekat::units;
+
+  grid_ = grids_manager->get_grid("Physics");
+  const auto& grid_name = grid_->name();
+
+  ncol_ = grid_->get_num_local_dofs();      // number of columns on this rank
+  nlev_ = grid_->get_num_vertical_levels(); // number of levels per column
+  nswbands_ = 14;                           // number of shortwave bands
+  nlwbands_ = 16;                           // number of longwave bands
```

Comment:
The appended underscore is my informal style for expressing that a variable or function is attached to a context (either it's a member of an object or a static variable within some namespace). EAMxx doesn't have a style guide, so you'll see other things, like Luca's Hungarian-notation-like `m_var` for member variables. I like these contextual "tags" because it makes it less confusing for the reader to figure out where the variable is coming from, and it prevents one from declaring a symbol that occludes a contextual variable.

The EAMxx team has discussed whether to have a style guide in the past, and it was decided to be unnecessary. As the code matures, we might reconsider whether to revisit this decision. There are many auto-formatters out there that eliminate the "holy wars" that often come with adopting styles, which changes the nature of the decision.

Reply:
It is one of the common ways to signal that the variable is a class member. That way, you can immediately tell if the var is local to the fcn or a class member. Common choices are `blah_` (this is google's guideline), `_blah`, `m_blah`, `M_blah`, `_M_blah` (this is what std lib uses)`. I like `m_blah` (m for member), which becomes `s_blah` if the member is static. But it is really a personal taste issue.